writeengine/splitter/we_filereadthread.cpp

```diff
@@ -200,37 +223,33 @@ bool WEFileReadThread::chkForListOfFiles(std::string& FileName)
     std::string aFileName = FileName;
 
     istringstream iss(aFileName);
+    ostringstream oss;
     size_t start = 0, end = 0;
     const char* sep = " ,|";
-
-    end = aFileName.find_first_of(sep);
+    ms3_status_st ms3status;
 
     do
     {
-        if (end != string::npos)
-        {
-            std::string aFile = aFileName.substr(start, end - start);
+        end = aFileName.find_first_of(sep, start);
+        std::string aFile = aFileName.substr(start, end - start);
+        if (aFile == "STDIN" || aFile == "stdin")
+            aFile = "/dev/stdin";
 
-            if (fSdh.getDebugLvl() > 2)
-                cout << "File: " << aFileName.substr(start, end - start) << endl;
+        if (fSdh.getDebugLvl() > 1)
+            cout << "Next Input File " << aFile << endl;
 
-            start = end + 1;
-            fInfileList.push_back(aFile);
-        }
-        else
+        if ((!doS3Import && access(aFile.c_str(), O_RDONLY) != 0) ||
+            (doS3Import && ms3_status(s3Connection, s3Bucket.c_str(), 
+                aFile.c_str(), &ms3status) != 0))
```

Comment:
consolidated the if-stmt and moved the file access check to here from we_sdlhandler.